### PR TITLE
[HUD] Add PR title to the page title

### PR DESCRIPTION
**Pull request by malfet**
Tried using `useEffect(()=> { document.title = "..."; }, []};` trick initially, but this does not seem to work reliable, so end up adding `DynamicTitle` component that defines a context(thank you ChatGPT for the hint on how to read React docs), and define `useSetTitle(title:string)` hook to update page title from the components

Also make title of top level HUD pages smaller

### CODE REVIEW

File: torchci/pages/hud/[repoOwner]/[repoName]/[branch]/[[...page]].tsx [hunk 1]
```diff
@@ -433,7 +433,7 @@ export default function Hud() {
   return (
     <>
       <Head>
-        <title>PyTorch CI HUD {title}</title>
+        <title>CI HUD {title}</title>
```
Comment (ZainRizvi) [inline torchci/pages/hud/[repoOwner]/[repoName]/[branch]/[[...page]].tsx:436]: This was needed! Could even do:

```suggestion
        <title>HUD {title}</title>
```

File: torchci/pages/[repoOwner]/[repoName]/commit/[sha].tsx [hunk 2]
```diff
@@ -67,6 +68,10 @@ export default function Page() {
       ? "ExecuTorch"
       : `${repoOwner}/${repoName}`;
 
+  useEffect(() => {
```
Comment (ZainRizvi) [inline torchci/pages/[repoOwner]/[repoName]/commit/[sha].tsx:71]: This one doesn't seem to update the title? Or am I looking in the wrong place?

**Review by ZainRizvi — APPROVED** [review]

**Review by huydhn — APPROVED** [review]

**Comment by huydhn** [comment]
I think this works, but only for PR page https://torchci-git-malfet-smarter-title-fbopensource.vercel.app/pr/126315.  Could we make this work for commit page too, i.e. https://torchci-git-malfet-smarter-title-fbopensource.vercel.app/pytorch/pytorch/commit/0c7601871472c468d84daa136d26e70c59f39594?

File: torchci/pages/[repoOwner]/[repoName]/commit/[sha].tsx (hunk 2 above)
Comment (malfet) [reply torchci/pages/[repoOwner]/[repoName]/commit/[sha].tsx:71]: It does, on line 72

**Pull request merged**